compiler/rustc_incremental/src/persist/load.rs

```diff
@@ -199,9 +199,14 @@ pub fn load_dep_graph(sess: &Session) -> DepGraphFuture {
     }))
 }
 
-pub fn load_query_result_cache(sess: &Session) -> OnDiskCache<'_> {
+/// Attempts to load the query result cache from disk
+///
+/// If we are not in incremental compilation mode, returns `None`.
+/// Otherwise, tries to load the query result cache from disk,
+/// creating an empty cache if it could not be loaded.
+pub fn load_query_result_cache(sess: &Session) -> Option<OnDiskCache<'_>> {
     if sess.opts.incremental.is_none() {
-        return OnDiskCache::new_empty(sess.source_map());
+        return None;
```

Comment:
`new_empty` should be cheap.

Reply:
Yes, but we want to skip running any operations on it (e.g. storing diagnostics, looking up query results). The alternative would be to insert checks for incremental mode into all of the methods in `OnDiskCache`